gcl_iam/controllers.py

```diff
@@ -174,3 +178,10 @@ def update(self, uuid, **kwargs):
         dm.update_dm(values=kwargs)
         dm.update()
         return dm
+
+
+class PolicyBasedCheckOtpController(PolicyBasedController):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self._introspection.get("otp_enabled"):
```

Comment:
I'd propose to move this check into _check_otp and maybe add argument `mandatory=False` (so if `mandatory=True` we may force otp even for accounts with disabled otp), in this case later we could set appropriate level of OTP checks in services.